### ISSUE TITLE
Not required (optional) fields in multipart/form-data don't have types in swagger

### ISSUE THREAD

**Issue by fluffybrain3** [luolingchun/flask-openapi3#126]
Environment:

- Python version: 3.12
- Operating system: Windows 11
- Flask version: 3.0.0
- flask-openapi3 version: 3.0.0

When request content is of type multipart/form-data every kind of field (FileStorage, bool, etc.) seems to be displayed just as a text field.
In the screenshot below you can see the flask-openapi3 3.0.0 version on the left side and on the right side there is 2.5.4 version with the same code.
![image](https://github.com/luolingchun/flask-openapi3/assets/151268312/633666a4-ccef-4367-b1cf-c495d10ecb01)

**Comment by luolingchun** [comment]
I can't reproduce your problem, please paste your code.

**Comment by fluffybrain3** [comment]
This worked in the previous version (doesn't work in this one)
```
class UploadFileForm(BaseModel):
    file: Optional[FileStorage]
    # other fields
```

It was changed by the pydantic-bump to
```
class UploadFileForm(BaseModel):
    file: Optional[FileStorage] = None
    # other fields
```
and still doesn't work.

Standard case when the field is required works fine for me:
```
class UploadFileForm(BasePydanticModel):
    file: FileStorage
    # other fields
```

So I am not able to declare a Form field which is not required according to the Pydantic docs:
https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields
because then there will be no field types in swagger.

**Comment by luolingchun** [comment]
`f7` can solve your problem.

![image](https://github.com/luolingchun/flask-openapi3/assets/22740403/f40e8edf-2e4a-4dc6-a7b5-a3880b4fa963)

**Comment by fluffybrain3** [comment]
I don't understand how f7 solves the problem.
I will try to explain in a different way.
It is not possible to declare a Form field which at the same time:
1. Is not required
2. Has correct type shown in swagger.
When you use f7 you only achieve the 1st point and the field in swagger is just a text input e.g. for the FileStorage field.

**Comment by luolingchun** [comment]
You misunderstood, use `file: FileStorage = None`, not `file: Any= None`.

**Comment by fluffybrain3** [comment]
Okay, so it seems like `file: FileStorage = None` is the only type that works the way you describe, but others don't.

```
class RequestForm(BaseModel):
    file: FileStorage = None
    some_text: str = None
    some_bool: bool = None
```

![image](https://github.com/luolingchun/flask-openapi3/assets/151268312/a25c583e-d384-45f8-b9c9-6dcef116c843)
![image](https://github.com/luolingchun/flask-openapi3/assets/151268312/c5aff979-dfae-4fe2-94bc-9ab85755213e)

**Comment by luolingchun** [comment]
I got it.

This issue as same as the #128 that will be fix by #129.

**Comment by luolingchun** [comment]
@fluffybrain3 It's worked for me!

```python
from pydantic import BaseModel

from flask_openapi3 import OpenAPI, FileStorage

app = OpenAPI(__name__)


class RequestForm(BaseModel):
    file: FileStorage = None
    some_text: str = None
    some_bool: bool = None


@app.post('/upload/file')
def upload_file(form: RequestForm):
    print(form)

    return {"code": 0, "message": "ok"}


if __name__ == '__main__':
    app.run(debug=True)

```


![image](https://github.com/luolingchun/flask-openapi3/assets/22740403/fbaaecad-6467-4aa3-b1ea-2a97be9f2bd5)

**Comment by fluffybrain3** [comment]
Hmmm, okay, now it's possible to declare a not required field in a Form this way.
But if you add the Optional type hint it doesn't work.
According to the Pydantic docs not required field should be declared with the Optional.
Without the Optional you cannot tick the "Send empty value" for e.g. int, bool types, because it will throw an error "Input should be a valid boolean, unable to interpret input".
```python
class RequestForm(BaseModel):
    file: Optional[FileStorage] = None
    some_text: Optional[str] = None
    some_bool: Optional[bool] = None
```
![image](https://github.com/luolingchun/flask-openapi3/assets/151268312/97a80408-8ea9-43b2-bdc2-8aec34210db3)

**Comment by luolingchun** [comment]
```python
class RequestForm(BaseModel):
    file: FileStorage = None
    some_text: str = None
    some_bool: Optional[bool] = None
```

Adding `Optional` will add an additional type `"type": "null"`.

I don't think [Swagger UI](https://github.com/swagger-api/swagger-ui) is ready for this yet. Or rather, there is no need for multi type support.

Therefore, the best solution currently is not to use `Optional` and not to send empty value.

```diff
"RequestForm": {
    "properties": {
        "file": {
            "format": "binary",
            "title": "File",
            "type": "string"
        },
        "some_bool": {
            "anyOf": [
                {
                    "type": "boolean"
                },
+               {
+                   "type": "null"
+               }
            ],
            "title": "Some Bool"
        },
        "some_text": {
            "title": "Some Text",
            "type": "string"
        }
    },
    "title": "RequestForm",
    "type": "object"
}
```

**Comment by fluffybrain3** [comment]
I mean, in the flask-openapi3 v2.5.4
```python
some_bool: Optional[bool] = None
some_bool: Optional[bool]
```
were parsed like this
```json
"some_bool": {
    "title": "Some Bool",
    "type": "boolean"
}
```
Version 3.0.0 changed it to
```json
"some_bool": {
    "anyOf": [
        {
            "type": "boolean"
        },
        {
            "type": "null"
        }
    ],
    "title": "Some Bool"
}
```
So was that change intentional?

**Comment by fluffybrain3** [comment]
I discovered another problem with Optional in v3 that didn't occur in v2.
```python
from typing import Optional
from uuid import UUID

from flask_openapi3 import OpenAPI
from pydantic import BaseModel

app = OpenAPI(__name__)


class RequestQuery(BaseModel):
    entity_ids: Optional[list[UUID]] = None


@app.post('/entities')
def upload_file(query: RequestQuery):
    print(query)

    return {"code": 0, "message": "ok"}


if __name__ == '__main__':
    app.run(debug=True)
```
![image](https://github.com/luolingchun/flask-openapi3/assets/151268312/7b8d7386-94d1-4ea2-ab7b-f3605ce39c50)

**Comment by luolingchun** [comment]
Pydantic v2 and OpenAPI 3.1 both follow [JSON schema specification draft 2020-12](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-4.2.1).

https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#data-types

https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-4.2.1

Here is an issue in Pydantic v2: https://github.com/pydantic/pydantic/issues/5394#issuecomment-1499529462


Overall, ` {"type": "null"} ` is a newly supported type.

`Option[bool]` and `Union[None, bool]` are the same, generating the same JSON schema.

```json
"anyOf": [
        {
            "type": "boolean"
        },
        {
            "type": "null"
        }
    ]
```

How about `Union[None, bool, str, int, ...]`? That will be generating more types, Swagger UI cannot handle it.

In summary, this is the reason for the problem.

Therefore, we need to avoid using multiple types in the `form` (application/form) but can use them in the `body` (application/json).
